packages/chain-adapters/src/cosmossdk/CosmosSdkBaseAdapter.ts

```diff
@@ -0,0 +1,221 @@
+    if (args.chainId && this.supportedChainIds.includes(args.chainId)) {
+      this.chainId = args.chainId
+    }
+    this.providers = args.providers
+  }
+
+  getType(): T {
+    throw new Error('Method not implemented.')
+  }
+
+  getChainId(): CAIP2 {
+    return this.chainId
+  }
+
+  getCaip2(): CAIP2 {
+    return this.chainId
+  }
+
+  async getAccount(pubkey: string): Promise<chainAdapters.Account<T>> {
+    try {
+      const caip = await this.getCaip2()
+      const { chain, network } = caip2.fromCAIP2(caip)
+      const { data } = await this.providers.http.getAccount({ pubkey })
+
+      return {
+        balance: data.balance,
+        caip2: caip,
+        // This is the caip19 for default asset on the chain (ATOM/OSMO/etc)
+        caip19: caip19.toCAIP19({
+          chain,
+          network,
+          assetNamespace: AssetNamespace.Slip44,
+          assetReference: '118'
+        }),
+        chain: this.getType(),
+        chainSpecific: {
+          sequence: data.sequence
+        },
+        pubkey: data.pubkey
+      } as chainAdapters.Account<T>
+    } catch (err) {
+      return ErrorHandler(err)
+    }
+  }
+
+  buildBIP44Params(params: Partial<BIP44Params>): BIP44Params {
+    return { ...ChainAdapter.defaultBIP44Params, ...params }
+  }
+
+  async getTxHistory(
+    input: chainAdapters.TxHistoryInput
+  ): Promise<chainAdapters.TxHistoryResponse<T>> {
+    try {
+      const { data } = await this.providers.http.getTxHistory({
+        pubkey: input.pubkey
+      })
+
+      throw new Error('Method not implemented.')
+      // return {
+      //   page: data.page,
+      //   totalPages: data.totalPages,
+      //   transactions: data.txs.map((tx) => ({
+      //     /*
+      //     missing properties from cosmos.api.Tx
+      //       status: string
+      //       from: string
+      //       to?: string
+      //       confirmations?: number
+      //      */
+      //     chain: caip2.fromCAIP2(this.getCaip2()).chain,
+      //     network: caip2.fromCAIP2(this.getCaip2()).network,
+      //     symbol: this.symbol
+      //   })),
+      //   txs: data.txs.length
+      // }
+    } catch (err) {
+      return ErrorHandler(err)
+    }
+  }
+
+  async buildSendTransaction(
+    tx: chainAdapters.BuildSendTxInput<T>
+  ): Promise<{ txToSign: chainAdapters.ChainTxType<T> }> {
+    throw new Error('Method not implemented.')
+  }
+
+  async getAddress(input: chainAdapters.GetAddressInput): Promise<string> {
+    // Method is implementation-specific
+    throw new Error('Method not implemented.')
+  }
+
+  async getFeeData(
+    input: Partial<chainAdapters.GetFeeDataInput<T>>
+  ): Promise<chainAdapters.FeeDataEstimate<T>> {
+    throw new Error('Method not implemented.')
+  }
+
+  async signTransaction(
+    signTxInput: chainAdapters.SignTxInput<chainAdapters.ChainTxType<T>>
+  ): Promise<string> {
+    // Method is implementation-specific
+    throw new Error('Method not implemented.')
+  }
```

Comment:
If all of these need to be implementation specific, we should just describe them as `abstract` function requiring the child class to implement